### PR TITLE
Remove warning about Logbook bug

### PR DESCRIPTION
**Pull request by msdousti**
The official documentation mentions a **major bug** for Logbook:

image

It refers to [this issue](https://github.com/zalando/logbook/issues/1216) on the Logbook project, which has been fixed by [this PR](https://github.com/zalando/logbook/pull/1538), and integrated into [Logbook 3.1.0](https://github.com/zalando/logbook/compare/3.0.0...3.1.0) on May 19, 2023.

As such, I'd like to suggest removing this waning from the documentation. Thanks!

### CODE REVIEW

**Review by sdelamo — APPROVED** [review]
I have updated logbook which we use in the test classpath to 3.6.0

**Comment by sdelamo** [comment]
Test `def 'h2c do not upgrade` fails after upgrading to the latest logbook. @yawkat could you check it out?

**Comment by yawkat** [comment]
ive merged 4.2.x and now the logbook test suite tests fail

**Comment by msdousti** [comment]
The `h2c do not upgrade` test is failing because of these lines:

https://github.com/micronaut-projects/micronaut-core/blob/0530ff704695dedee4d614f2e6cb2e055a6128dc/test-suite/src/test/groovy/io/micronaut/docs/netty/LogbookNettyServerCustomizerSpec.groovy#L338-L339

As explained in the description of the PR, with new Logbook version (3.1 onwards), this bug is resolved, so the "second" `bar` should now be visible in the logs. I'll adapt the test.

For some other reason, another test `tls alpn http 2` started to fail after merging `4.2.x` onto this branch. I bumbed Logbook version on `4.2.x` itself and the test passed. ~~A bit strange...~~

**PS**
The test `tls alpn http 2` failed because of the `@Requires` added in 4fefb40. For the moment, "fixed" by removing it. Please see 6920568857a90b872e7c201f5c83497dd6970ada.

File: test-suite/src/test/groovy/io/micronaut/docs/netty/LogbookNettyServerCustomizerSpec.groovy [hunk 1]
```diff
@@ -480,7 +479,6 @@ class LogbookNettyServerCustomizerSpec extends Specification {
         }
     }
 
-    @Requires(property = 'spec.name', value = 'LogbookNettyServerCustomizerSpec')
```
Comment (yawkat) [inline test-suite/src/test/groovy/io/micronaut/docs/netty/LogbookNettyServerCustomizerSpec.groovy]: this should stay

**Review by yawkat — CHANGES_REQUESTED** [review]

File: test-suite/src/test/groovy/io/micronaut/docs/netty/LogbookNettyServerCustomizerSpec.groovy (hunk 1 above)
Comment (yawkat) [reply test-suite/src/test/groovy/io/micronaut/docs/netty/LogbookNettyServerCustomizerSpec.groovy]: let me check why `tls alpn http 2` is fixed by this

**Comment by yawkat** [comment]
@msdousti can you sign the CLA?

**Comment by msdousti** [comment]
@yawkat Signed.

**Comment by yawkat** [comment]
Thanks!

**Pull request merged**